### PR TITLE
Ternary axis layer

### PR DESCRIPTION
**Pull request by etpinard**
Implements https://github.com/plotly/plotly.js/pull/1871 for ternary subplots

### CODE REVIEW

**Comment by etpinard** [comment]
I still have to add a few jasmine tests. I'm opening this PR early to tag it under the `v1.30.0` milestone.

File: src/plots/ternary/ternary.js [hunk 1]
```diff
@@ -95,50 +96,75 @@ proto.makeFramework = function() {
...
 
-    var grids = _this.plotContainer.select('.grids').selectAll('g.grid')
-        .data(['agrid', 'bgrid', 'cgrid']);
-    grids.enter().append('g')
-        .attr('class', function(d) { return 'grid ' + d; })
-        .each(function(d) { _this.layers[d] = d3.select(this); });
-
-    _this.plotContainer.selectAll('.backplot,.grids')
-        .call(Drawing.setClipUrl, clipId);
+    toplevel.order();
```
Comment (etpinard) [inline src/plots/ternary/ternary.js]: see https://bl.ocks.org/mbostock/1440766

File: src/plots/ternary/ternary.js [hunk 2]
```diff
@@ -95,50 +96,75 @@ proto.makeFramework = function() {
...
-    var plotLayers = [
-        'draglayer',
-        'plotbg',
-        'backplot',
-        'grids',
-        'frontplot',
-        'aaxis', 'baxis', 'caxis', 'axlines'
-    ];
+
+    var plotLayers = ['draglayer', 'plotbg', 'backplot', 'grids'];
```
Comment (etpinard) [inline src/plots/ternary/ternary.js]: The handful of lines below are a little bit verbose. If anyone knows a cleaner / better way to do this, let me know.

**Comment by etpinard** [comment]
Looks like CircleCI is down for the moment. Nonetheless, `type: reviewable`.

File: src/plots/ternary/ternary.js (hunk 2 above)
Comment (alexcjohnson) [inline src/plots/ternary/ternary.js]: dunno if it's worthwhile - shorter but not necessarily cleaner - but you could do something like
```js
var plotOverLayers = ['frontplot'];
['a','b','c'].forEach(function(axLetter) {
  var axName = axLetter + 'axis';
  var isBelow = ternaryLayout[axName].layer === 'below';
  (isBelow ? plotLayers : plotOverLayers).push(axName, axLetter + 'line');
});
[].push.apply(plotLayers, plotOverLayers);
```

**Comment by etpinard** [comment]
cc @geocosmite

**Comment by alexcjohnson** [comment]
The `cliponaxis: true` trace with `axis.layer: 'below traces'` in the new test image looks funny, how the markers cover up half the axis line. But I think trying to solve that - moving the edge of the clip path back to the edge of the axis line, for example - would cause more problems than it solves. This combination doesn't have a use case that's obvious to me, but if you moved the clip path back it would make a subpixel gap so the more common use case `cliponaxis: true` and `axis.layer: 'above traces'` would look bad.

So lets keep it as is. No more comments, the one above is arguably not even an improvement 🙈
💃

File: src/plots/ternary/ternary.js (hunk 1 above)
Comment (alexcjohnson) [inline src/plots/ternary/ternary.js]: dunno if `selection.order` wasn't in d3 back when I started using it or I just couldn't find it... but we need to use this maaaany more places in our code!

**Pull request merged**